arrow-json/src/reader/mod.rs

```diff
@@ -189,10 +197,51 @@ impl ReaderBuilder {
             batch_size: 1024,
             coerce_primitive: false,
             strict_mode: false,
+            is_field: false,
             schema,
         }
     }
 
+    /// Create a new [`ReaderBuilder`] with the provided [`FieldRef`]
```

Comment:
```suggestion
    /// Create a new [`ReaderBuilder`] that will parse JSON values with a root schema of [`FieldRef`].
```

Perhaps we can add a note in `new()` that says it does require the root to be an object like `{..}`.

I wonder if `new_from_field` might be a more descriptive name (as this isn't making a new Field, it is making a new reader that reads data with the type on the field